arangod/Aql/RegisterPlan.h

```diff
@@ -119,12 +119,15 @@ struct RegisterPlanT final : public std::enable_shared_from_this<RegisterPlanT<T
   // have the same length:
   std::vector<RegisterCount> nrRegs;
 
+  RegisterCount nrConstRegs = 0;
+
   // We collect the subquery nodes to deal with them at the end:
   std::vector<T*> subqueryNodes;
 
   /// @brief maximum register id that can be assigned, plus one.
   /// this is used for assertions
-  static constexpr RegisterId MaxRegisterId = RegisterId{1000};
+  static constexpr RegisterId MaxRegisterId = RegisterId(RegisterId::maxRegisterId);
+  // TODO - remove MaxRegisterId in favor of RegisterId::maxRegisterId
```

Comment:
What is keeping us from doing this now?

Reply:
Nothing really, but a quick search shows that there are still 139 usages of `MaxRegisterId` and I was a bit hesitant to make this change in this PR so that it does not become unnecessarily big. But if you prefer I am happy to do it now.

Reply:
Ok, if you haven't start it yet, then let's do it later.
However, it looks like a trivial search&replace, so we should do it after this PR has been merged.